grizzly/users/messagequeue.py

```diff
@@ -180,8 +203,30 @@ def __init__(self, *args: Tuple[Any, ...], **kwargs: Dict[str, Any]) -> None:
 
 
     def request(self, request: RequestTask) -> None:
+
+        # Parse the endpoint to extract queue name / expression parts
+        queue_name = request.endpoint
+        expression: Optional[str] = None
+
+        # Remove any 'queue:' prefix
+        queue_name = resub(r'^\s*queue:\s*', '', queue_name)
+
+        if ',' in queue_name:
+            queue_name, expression = [x.strip() for x in queue_name.split(',')]
+            if not expression.startswith('expression:'):
+                logger.error(f'Expression part in endpoint needs to have "expression:" in it: {request.endpoint}')
+                raise StopUser()
+            # Remove 'expression:' prefix and keep the value
+            expression = resub(r'\s*expression:\s*(.+?)\s*$', r'\1', expression)
+
+        # Keep queue name part in request.endpoint
+        request.endpoint = queue_name
```

Comment:
the request can be executed more than once (depending on how many iterations the scenario has), and with this it will have been changed next time it will execute.